src/charm.py

```diff
@@ -104,22 +95,26 @@ def _on_config_changed(self, event: ops.ConfigChangedEvent):
                 )
             )
 
-            # FIXME: remove after kafka_client
-            self.config_manager._update_config(
-                uris=self.config.bootstrap_servers,
-                username=self.config.username,
-                password=self.config.karapace_password,
-            )
-
+            # Config is different, apply changes to file
             self.config_manager.generate_config()
 
-        if not self.workload.active():
-            self.workload.start()
-        else:
-            self.workload.restart()
+            # Restart so changes take effect
+            logger.info("Acquiring lock from _on_config_changed...")
```

Comment:
**nitpick:** I think this log can be removed. I have removed it in Kafka now too, I think it was leftover debugging from ages ago.